modules/videoio/src/cap_android_mediandk.cpp

```diff
@@ -154,30 +153,58 @@ class AndroidMediaNdkCapture : public IVideoCapture
         if (buffer.empty()) {
             return false;
         }
-        Mat yuv(frameHeight + frameHeight/2, frameWidth, CV_8UC1, buffer.data());
+
+        Mat yuv(frameHeight + frameHeight/2, frameStride, CV_8UC1, buffer.data());
```

Comment:
You need to use this constructor:
```
Mat (Size size, int type, void *data, size_t step=AUTO_STEP);
```
where step=stride.

Reply:
I think I already tried it without luck, but I'll double check it.